src/main/java/com/google/sps/servlets/AuthServlet.java

```diff
@@ -89,7 +83,7 @@ private String convertToJson(UserAuth userAuth) {
   /**
    * If the user is not in database already, add them.
    */
-  private void addUserToDatabase(String email) {
+  public void addUserToDatabase(String email) {
```

Comment:
In the following CL, I fix the Query by adding a filter that only gets the Entity with the `email` in question. However, this could also be avoided if I simply put the object into Datastore, as it seems that takes care of these concerns. I'll test this right now.

Reply:
I tried the change, but in this instance, Datastore stores **both** Entity objects, and doesn't test that it already exists. I think I know why this is.

When you create the Entity object, it isn't assigned a **key** or **id**. However, when you add it to Datastore, it is then assigned these attributes. Therefore, when I use the `datastore.put(...)` method without checking if the User is in Datastore, it adds another Entity object -- one with different **key** and **id** properties, but the same **email** attribute. Since we only need / want one Entity object with this **email** attribute in Datastore, it is necessary to check whether the User is already in the Datastore.

Reply:
Ah, I see, okay. Thanks for checking that out